### PR TITLE
Use Mesos 0.26 version of NetworkInfo

### PR DESCRIPTION
**Pull request by Symmetric**
In MESOS-3788, the NetworkInfo message was updated to clarify
the structure of how IPs map onto interfaces.

Update mesos-dns to use the new NetworkInfo.

### CODE REVIEW

**Comment by Symmetric** [comment]
@jdef can you give this a review please? Think it's good to go per our discussion here https://github.com/Symmetric/mesos-dns/commit/d66fa869e4bfe31fe941dd6c9a7b69eb3a70e7de#commitcomment-14390594

**Comment by sargun** [comment]
@Symmetric Can you rebase on master, please?

**Comment by Symmetric** [comment]
@sargun rebased. I'm seeing failures both on master and this branch; not sure if there's a problem with my env or the code.

**Comment by jdef** [comment]
lgtm

**Comment by jdef** [comment]
metalinter is complaining:
```
records/state/state.go:68:6:warning: exported type IPAddress should have comment or be unexported (golint)

records/state/state.go:135:11:warning: don't use underscores in Go names; range var ip_address should be ipAddress (golint) gometalinter --cyclo-over=12 --tests --deadline=60s ./... returned exit code 1
```

File: records/state/state.go [hunk 1]
```diff
@@ -123,12 +127,14 @@ var sources = map[string]func(*Task) []string{
...
 // networkInfoIPs returns IP addresses from a given Task's
-// []Status.ContainerStatus.[]NetworkInfos.IPAddress
+// []Status.ContainerStatus.[]NetworkInfos.[]IPAddresses.IPAddress
 func networkInfoIPs(t *Task) []string {
 	return statusIPs(t.Statuses, func(s *Status) []string {
 		ips := make([]string, len(s.ContainerStatus.NetworkInfos))
-		for i, netinfo := range s.ContainerStatus.NetworkInfos {
-			ips[i] = netinfo.IPAddress
+		for _, netinfo := range s.ContainerStatus.NetworkInfos {
+			for _, ip_address := range netinfo.IPAddresses {
```
Comment (jdef) [inline records/state/state.go]: ip_address is not an idiomatic name

File: records/state/state.go [hunk 2]
```diff
@@ -62,6 +62,10 @@ type ContainerStatus struct {
 // NetworkInfo holds a network address resolution as defined in the
 // /state.json Mesos HTTP endpoint.
 type NetworkInfo struct {
+	IPAddresses []IPAddress `json:"ip_addresses,omitempty"`
+}
+
+type IPAddress struct {
```
Comment (jdef) [inline records/state/state.go]: metalinter wants docs for this

**Comment by Symmetric** [comment]
@jdef done -- want me to squash now that the tests are green?

**Comment by jdef** [comment]
yes please squash

**Comment by Symmetric** [comment]
@jdef done.

**Comment by jdef** [comment]
:shipit: @sargun i'll leave the final merge up to you

**Comment by jdef** [comment]
We need to make sure that this mesos-dns change is timed with the DCOS release that includes v0.26.

**Comment by sargun** [comment]
What's the backwards compatibility story?

**Comment by jdef** [comment]
It's not compat w/ 0.25 -- with this change mesos-dns will no longer see
v0.25's NetworkInfo.ip_address. Probably makes sense to tag mesos-dns with
a v0.25-compat label or something, before this change is merged. Definitely
worth a release note for the next build.

Talked with the DCOS folks and since mesos/mesos-dns are upgraded together
people generally aren't worried about compat problems.

**Comment by sargun** [comment]
@jdef I'd like to start the release notes for v-next, and add a note about compatibility into the README. Since I can't easily put this Symmetric:mesos-0.26-network-info, so I'm going to cut a v-next branch where we can merge this branch into, and then merge that into master. Mostly, as to get this into an atomic merge.

**Comment by sargun** [comment]
See: v-0.26

**Comment by jdef** [comment]
sgtm

**Comment by jdef** [comment]
this PR is likely a blocker for the network-modules folks. we need to
coordinate with them.

**Comment by Symmetric** [comment]
Closing in favour of #355 (rebased onto v-0.26 instead of master)

**Pull request closed without merging**